### PR TITLE
[BugFix] fix bug that repetition_penalty is not effective in asynchronous scheduling

### PR DESCRIPTION
**Pull request by SumrainChan**
### What this PR does / why we need it?

To fix the [issue:7769](https://github.com/vllm-project/vllm-ascend/issues/7769)

The GPUModelRunner updates the output_token_ids of sampling_metadata before each sample. However, the NPUModelRunner does not perform this step. As a result, the generated token cannot be obtained during apply_repetition_penalties, and the repetition_penalty is not effective when it is greater than 1.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

- vLLM version: v0.18.0
- vLLM main: https://github.com/vllm-project/vllm/commit/35141a7eeda941a60ad5a4956670c60fd5a77029

### CODE REVIEW

**Comment by realliujiaxu** [comment]
Thanks for your contribution! Let's merge it after CI pass.

**Comment by realliujiaxu** [comment]
LGTM. Could you add a test case for this?

**Comment by SumrainChan** [comment]
> Thanks for this bugfix! I have 2 suggestions.
> 
> 1. I checked the code and find this bug is introduced because we didn't adapt async scheduler + penalties before. And this is done in vllm by [vllm-project/vllm#26467](https://github.com/vllm-project/vllm/pull/26467).
> 
> And your fix seems lack of the logic of passing `logitsprocs_need_output_token_ids` by the `InputBatch`
> 
> 2. could you please add an test case of async scheduler + penalties scenario to prevent the problem from recurring?


Thanks for your suggestions. I will add logitsprocs_need_output_token_ids and test cases later.
